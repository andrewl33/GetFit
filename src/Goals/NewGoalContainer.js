import React, { Component } from 'react';
import { FlatList, StyleSheet, Text, View, Button, Alert } from 'react-native';
import { _getNewGoals, _addNewGoal } from '../Storage/GoalsStorage';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    alignSelf: 'stretch',
    marginTop: 50,
  },
  header: {
    fontSize: 35,
  },
  flatList: {
    alignSelf: 'stretch',
  },
  button: {
    margin: 30,
  },
});

export default class NewGoalsContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      items: [],
    };
  }

  // TODO: remove eventually
  AlertText = () => {
    Alert.alert('Goal Added!');
  };

  componentDidMount = async () => {
    const objs = await _getNewGoals(); // comes in as obj of objs
    // convert object of objects into array of objects
    const newGoals = Object.keys(objs).map(key => {
      const ar = objs[key];
      ar.selected = false;
      return ar;
    });
    this.setState({ items: newGoals });
  };

  showGoalDetails = ({ item }) => {
    Alert.alert(
      `Goal:  ${item.endAmount} ${item.unitType}\n When:  ${item.createDate} to ${item.endDate}`
    );
  };

  selectItem = ({ item }) => {
    const newItem = { ...item };
    newItem.selected = !newItem.selected;

    this.setState(prevState => {
      const newState = [...prevState.items];
      const index = prevState.items.indexOf(item);
      newState[index] = newItem;
      return {
        items: newState,
      };
    });
  };

  _keyExtractor = item => String(item.id);

  _renderItem = ({ item }) => (
    <View style={styles.button}>
      <Button onPress={() => this.showGoalDetails({ item })} title={item.title} />
      <Button
        onPress={() => this.selectItem({ item })}
        title={item.selected ? 'Selected' : 'Select'}
      />
    </View>
  );

  render() {
    const { items } = this.state;
    return (
      <View style={styles.container}>
        <Text style={styles.header}>Select A Goal</Text>
        <FlatList
          style={styles.flatList}
          data={items}
          keyExtractor={this._keyExtractor}
          renderItem={this._renderItem}
        />
        <View style={styles.button}>
          <Button onPress={() => this.AlertText()} title="Add Selected Goals">
            Add New Goals
          </Button>
        </View>
      </View>
    );
  }
}
