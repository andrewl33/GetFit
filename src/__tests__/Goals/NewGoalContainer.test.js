import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import NewGoalContainer from '../../Goals/NewGoalContainer';

describe('NewGoalContainer', () => {
  it('renders', () => {
    const tree = renderer.create(<NewGoalContainer />);
    expect(tree).toMatchSnapshot();
  });
});
