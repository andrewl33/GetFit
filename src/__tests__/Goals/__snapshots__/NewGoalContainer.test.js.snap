// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewGoalContainer renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "flex": 1,
      "justifyContent": "center",
      "marginTop": 50,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 35,
      }
    }
  >
    Select A Goal
  </Text>
  <View
    style={
      Object {
        "margin": 30,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Add Selected Goals
        </Text>
      </View>
    </View>
  </View>
</View>
`;
